app/src/main/java/com/example/happybirthday/unit1/UnitOneActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.happybirthday.unit1
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import com.example.happybirthday.R
+
+class UnitOneActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val rollButton: Button = findViewById(R.id.button)
+        rollButton.setOnClickListener { rollDice() }
+
+    }
+
+    /**
+     * Method create Dice object and add random number to the screen.
+     */
+    private fun rollDice() {
+        val dice = Dice(6)
+        val diceRoll = dice.roll()
+        val imageView: ImageView = findViewById(R.id.imageView2)
+        val textView: TextView = findViewById(R.id.textView1)
+        val drawableResource = when (diceRoll) {
+            1 -> R.drawable.dice_1
+            2 -> R.drawable.dice_2
+            3 -> R.drawable.dice_3
+            4 -> R.drawable.dice_4
+            5 -> R.drawable.dice_5
+            else -> R.drawable.dice_6
+        }
+
+        imageView.setImageResource(drawableResource)
+        textView.setText(R.string.signature_text)
+        var text: String = textView.text.toString()
+        text = "$text $diceRoll"
```

Comment:
we better use strings with placeholders like `"Your number %d"`. See https://developer.android.com/guide/topics/resources/string-resource#formatting-strings.
Also, if you need to get a string from `strings.xml` and you are in `Activity`, use getString(R.string.your_string)